components/hdw-tft/hdw-tft.c

```diff
@@ -354,7 +360,7 @@ void enableTFTBacklight(void)
             .duty       = 255, // Disable to start.
         };
         ESP_ERROR_CHECK(ledc_channel_config(&ledc_config_backlight));
-        setTFTBacklightBrightness(CONFIG_TFT_DEFAULT_BRIGHTNESS);
+        setTFTBacklightBrightness(tftBacklightIntensity);
```

Comment:
If `initTFT()` is called before `setTFTBacklightBrightness()`, I think this value is undefined. That's not great. It should have a default value.

Reply:
@dylwhich if I'm reading this right, it needs a little fix.

Reply:
Ah yeah, I think so. Would it be terrible to load it from NVS here? Or I can just give the variable a default.

The actual issue I want to fix is that we call `readAllSettings()` pretty much first thing on boot, which tries to set some settings that just get overwritten by initializing the components right after. I tried to just call `readAllSettings()` again after everything was initialized, but in the case of LED brightness that still produced a noticeable flash.

Reply:
Ah, if that's the issue, then the various initializers should get passed the settings from `app_main()`. It's cleaner than creating a dependency on NVM in all configurable components. And maybe `readAllSettings()` should stop setting stuff before it's initialized...

Reply:
Yeah, oof, I really punted `readAllSettings()`. It should _only_ read, and pass:
* Volumes into `initBuzzer()`
* Brightness into `initTFT()`
* Brightness into `initLeds()`

Alternatively, it could read only, then have `app_main()` set settings after initializing components, but I dunno if that would solve your noticeable flash.

Reply:
Oh wait, I actually started adding the settings as parameters. I just forgot to actually use `brightness` here, oops! I'll check on the other ones though.

Reply:
I think that should do it for the rest of them!